sudospawner/spawner.py

```diff
@@ -94,6 +94,19 @@ def start(self):
 
     @gen.coroutine
     def _signal(self, sig):
+        if sig == 0:
+            # short-circuit existence check without invoking sudo
+            try:
+                os.kill(self.pid, sig)
+            except ProcessLookupError:
+                # No such process
+                return False
+            except PermissionError:
+                # When running hub with reduced permissions,
+                # we won't have permission to send signals, even 0.
+                # PermissionError means process exists.
```

Comment:
> What impact will that have on hub behavior?

The same as in the regular local process case - the Hub will think the user's server is running (it isn't). There should be an additional check for whether the server is actually up and responsive, and then it will clear this state if there is actually no server running at the stored URL.